active_standby.py

```diff
@@ -45,70 +45,107 @@ def refresh_login():
+        jmessage = json.loads(message)
+        message = jmessage["imdata"][0]["ethpmAggrIf"]['attributes']
+
+        for sub_id in jmessage['subscriptionId']:
+
+            active_port_count = 0
+            if 'numActivePorts' in message:
+                active_port_count = int(message['numActivePorts'])
+    
+            if sub_id == active_standby.standby_sub_id:
+                # the event was on the subscription for the standby link, someone unshut it, lets stop that 
+    
+                # when the standby is unshut, a message is received with operStQual == 'port-channel-members-down'.
+                # then after some time if the portchannel comes up the 'numActivePorts' >= 0
+                # we'll try catch it on the first message, otherwise wait for following message
+    
+                being_unshut = False 
+                if 'status' in message and 'operStQual' in message:
+                    if message['status'] == 'modified' and message['operStQual'] == 'port-channel-members-down':
+                        being_unshut = True
+    
+                if active_port_count > 0 or being_unshut:
+                    # shutdown the standby (by blacklisting)
+                    query = "https://{apic_address}/api/node/mo/uni/fabric/outofsvc.json".format(apic_address=apic_address)
+                    data = {
+                        "fabricRsOosPath": {
+                            "attributes": {
+                                "tDn": "{port_channel_policy}".format(
+                                    port_channel_policy=active_standby.port_channel_policy_standby),
+                                "lc": "blacklist"
+                            }
                         }
                     }
-                }
-
-                resp = s.post(query, json=data, verify=False)
-                if not resp.ok:
-                    exit("Failed to flip to Standby Port Channel - CRITICAL ERROR")
-                else:
-                    print "Flipped to Standby Port Channel."
-
-                # shut the active (by blacklisting)
-                query = "https://{apic_address}/api/node/mo/uni/fabric/outofsvc.json".format(apic_address=apic_address)
-                data = {
-                    "fabricRsOosPath": {
-                        "attributes": {
-                            "tDn": "{port_channel_policy}".format(
-                                port_channel_policy=active_standby.port_channel_policy_active),
-                            "lc": "blacklist"
+    
+                    resp = s.post(query, json=data, verify=False)
+                    if not resp.ok:
+                        exit("Failed to shut Standby Port Channel - CRITICAL ERROR")
+                    print "Shutdown Standby Port Channel. "
+        
+            if sub_id == active_standby.active_sub_id:
+                # the event was on the subscription for the active link
+                try:
```

Comment:
Can we pull this `try:` statement a bit higher up to line 48. This should help catch any errors Key or Value errors gracefully within this entire `on_message` function.